### PR TITLE
Add support for BPF_PROG_TYPE_CGROUP_SOCKOPT

### PR DESCRIPTION
**Pull request by nak3**
This patch adds support for `BPF_PROG_TYPE_CGROUP_SOCKOPT`.

close https://github.com/aya-rs/aya/issues/219

### CODE REVIEW

**Comment by nak3** [comment]
https://github.com/nak3/aya-showcase/tree/main/cgroup-sockopt is the demonstration for this PR.

File: aya/src/programs/cgroup_sockopt.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    programs::{
+        bpf_attach_type, define_link_wrapper, load_program, FdLink, Link, OwnedLink,
+        ProgAttachLink, ProgramData, ProgramError,
+    },
+    sys::{bpf_link_create, bpf_prog_attach, kernel_version},
+};
+
+/// A program that can be used to get or set options on sockets.
+///
+/// [`SockAddr`] programs can be used to inspect or modify socket addresses passed to
```
Comment (alessandrod) [inline aya/src/programs/cgroup_sockopt.rs:19]: Copypasta from SockAddr detected 🚨

**Review by alessandrod — CHANGES_REQUESTED** [review]
You're on a roll!

Remove the copypasta but otherwise great stuff!

File: aya/src/programs/cgroup_sockopt.rs (hunk 1 above)
Comment (vadorovsky) [inline aya/src/programs/cgroup_sockopt.rs:19]:
```suggestion
/// [`CgroupSockopt`] programs can be used to inspect or modify socket addresses passed to
```
?

File: aya/src/programs/cgroup_sockopt.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// use std::fs::File;
+/// use std::convert::TryInto;
+/// use aya::programs::CgroupSockopt;
+///
+/// let file = File::open("/sys/fs/cgroup/unified")?;
+/// let program: &mut CgroupSockopt = bpf.program_mut("cgroup_sockopt").unwrap().try_into()?;
+/// program.load()?;
+/// program.attach(file)?;
+/// # Ok::<(), Error>(())
```
[the remark this answers is not in the document]
Comment (nak3) [reply aya/src/programs/cgroup_sockopt.rs]: I thought it is a nice idea but it seems CI fails with it https://github.com/aya-rs/aya/runs/6524168974?check_suite_focus=true :sweat_smile:

**Comment by nak3** [comment]
Thank you so much! Updated.

File: aya/src/programs/cgroup_sockopt.rs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        bpf_attach_type, define_link_wrapper, load_program, FdLink, Link, OwnedLink,
+        ProgAttachLink, ProgramData, ProgramError,
+    },
+    sys::{bpf_link_create, bpf_prog_attach, kernel_version},
+};
+
+/// A program that can be used to get or set options on sockets.
+///
+/// [`CgroupSockopt`] programs can be attached to a cgroup and will be called every
+/// time a process executes getsockopt or getsockopt system call.
```
Comment (alessandrod) [inline aya/src/programs/cgroup_sockopt.rs:20]: getsockopt or setsockopt

**Review by alessandrod — APPROVED** [review]
small doc typo

File: aya/src/programs/cgroup_sockopt.rs (hunk 3 above)
Comment (nak3) [reply aya/src/programs/cgroup_sockopt.rs]: Thank you! Updated.

**Pull request merged**